src/main/java/org/sonarlint/intellij/core/SonarQubeEventNotifications.java

```diff
@@ -0,0 +1,140 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.core;
+
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationListener;
+import com.intellij.notification.NotificationType;
+import com.intellij.openapi.components.AbstractProjectComponent;
+import com.intellij.openapi.project.Project;
+import com.intellij.util.messages.MessageBusConnection;
+import java.time.ZonedDateTime;
+import org.sonarlint.intellij.config.global.SonarQubeServer;
+import org.sonarlint.intellij.config.project.SonarLintProjectSettings;
+import org.sonarlint.intellij.config.project.SonarLintProjectState;
+import org.sonarlint.intellij.exception.InvalidBindingException;
+import org.sonarlint.intellij.messages.ProjectConfigurationListener;
+import org.sonarlint.intellij.util.SonarLintUtils;
+import org.sonarsource.sonarlint.core.client.api.common.NotificationConfiguration;
+import org.sonarsource.sonarlint.core.client.api.connected.ServerConfiguration;
+import org.sonarsource.sonarlint.core.client.api.notifications.SonarQubeNotification;
+import org.sonarsource.sonarlint.core.client.api.notifications.SonarQubeNotificationListener;
+import org.sonarsource.sonarlint.core.client.api.notifications.SonarQubeNotificationTime;
+import org.sonarsource.sonarlint.core.notifications.SonarQubeNotifications;
+
+public class SonarQubeEventNotifications extends AbstractProjectComponent {
+  public static final String GROUP_SONARQUBE_EVENT = "SonarLint: SonarQube Events";
+
+  private final ProjectBindingManager bindingManager;
+  private final SonarLintProjectState projectState;
+  private final SonarLintProjectSettings projectSettings;
+  private final EventListener eventListener;
+  private final SonarQubeNotificationTime notificationTime;
+  private final MessageBusConnection busConnection;
+
+  public SonarQubeEventNotifications(Project project, ProjectBindingManager bindingManager, SonarLintProjectState projectState,
+    SonarLintProjectSettings projectSettings) {
+    super(project);
+    this.bindingManager = bindingManager;
+    this.projectState = projectState;
+    this.projectSettings = projectSettings;
+    this.eventListener = new EventListener();
+    this.notificationTime = new ProjectNotificationTime();
+    this.busConnection = project.getMessageBus().connect(myProject);
+  }
+
+  @Override
+  public void projectOpened() {
+    register(projectSettings);
+    busConnection.subscribe(ProjectConfigurationListener.TOPIC, settings -> {
+      // always reset notification date, whether bound or not
+      projectState.setLastEventPolling(ZonedDateTime.now());
+      register(settings);
+    });
+  }
+
+  private void register(SonarLintProjectSettings settings) {
+    SonarQubeNotifications.get().remove(eventListener);
```

Comment:
This line appears twice in the file. As it's effectively the logical opposite of `register`, I would extract it to a method `unregister`.